### DISCUSSION TITLE
Run Carla Server on Linux and Client on Windows

### DISCUSSION THREAD

**Discussion by aakashsehgal** [carla-simulator/carla#5655] [General]
I installed Debian Carla 0.9.12 on Ubuntu 20.04. I can start the Carla server and also connect to the python client "manual_control.py".

Now i want to connect to this carla server via a Windows client. What do i need to install on the windows side and how can i run this configuration ?

**Comment by RaviBeagle** [comment]
It should be pretty easy as CARLA natively supports Client Server architecture
CARLA communicates over ports 2000 and 2001 by default I think. The Windows client and Linux server should be able to communicate with each other over the ports. On the Windows client system you would need the same CARLA version installation.
Most of the client programs have the command line arguments to specify the host and port like example:

```
    argparser.add_argument(
        '--host',
        metavar='H',
        default='127.0.0.1',
        help='IP of the host server (default: 127.0.0.1)')
    argparser.add_argument(
        '-p', '--port',
        metavar='P',
        default=2000,
        type=int,
        help='TCP port to listen to (default: 2000)')
```
Or you might need to modify the code otherwise. You could also specify the server to listen to on different ports.

**Reply by NaveedBhuiyan** [reply]
Has anyone implemented something like this? . I tried but still doesnt work. I have carla server running on my windows 10 and I am trying to connect to it through a docker container running linux with all the python and carla dependancies installed. however I keep on getting some sort of lib error.
According to this : https://carla.readthedocs.io/en/0.9.13/build_system/
It should not be possible since the carla.exe was built using platform dependant c++ libraries. Please let me know if you guys can find a way

**Reply by KaguraTart** [reply]
i tried to modify the server host， but it doesnt work. i tried to modify the source code about rpc and endpoint.h， Do not work too. Then I tried using ip 0.0.0.0 to connect server and  client, I  found that the ip and port is working, but  client still can’t connect the Carla  server

**Comment by apcial1** [comment]
I'd like to seek guidance on how to install Carla on a server. I encountered an issue while attempting to install Carla version 0.9.10.1, where it stops immediately after running with the following message:

4.24.3-0+++UE4+Release-4.24 518 0 Disabling core dumps. (carla) carlauser@notebook-devenviron-0110-182655-1oignbn-notebook-0:/workspace/algorithm/carla$

Any assistance or advice on resolving this issue would be greatly appreciated.

**Reply by Arch-suzuki-MB** [reply]
@apcial1 Do you try to install it on a remote server and what is the os of the server is it linux ubuntu and witch version of ubuntu if already do this can you please share and thank you

**Reply by KaguraTart** [reply]
I have done these things by running carla in ubuntu20.04 as server, and in the same local networks I run the carla client which can searched and got connected the unbuntu20.04 carla server.
Maybe you need make sure your local network close the firewall and devices can search each other.
